src/System.Private.CoreLib/shared/System/MemoryExtensions.cs

```diff
@@ -22,6 +22,181 @@ namespace System
     /// </summary>
     public static partial class MemoryExtensions
     {
+        /// <summary>
+        /// Removes all leading and trailing occurrences of a specified element.
+        /// </summary>
+        /// <param name="span">The source span from which the element is removed.</param>
+        /// <param name="trimElement">The specified element to look for and remove.</param>
+        public static ReadOnlySpan<T> Trim<T>(this ReadOnlySpan<T> span, T trimElement)
+            where T : IEquatable<T>
+            => TrimImpl(ref span, trimElement, true, true);
+
+        /// <summary>
+        /// Removes all leading occurrences of a specified element.
+        /// </summary>
+        /// <param name="span">The source span from which the element is removed.</param>
+        /// <param name="trimElement">The specified element to look for and remove.</param>
+        public static ReadOnlySpan<T> TrimStart<T>(this ReadOnlySpan<T> span, T trimElement)
+            where T : IEquatable<T>
+            => TrimImpl(ref span, trimElement, true, false);
+
+        /// <summary>
+        /// Removes all trailing occurrences of a specified element.
+        /// </summary>
+        /// <param name="span">The source span from which the element is removed.</param>
+        /// <param name="trimElement">The specified element to look for and remove.</param>
+        public static ReadOnlySpan<T> TrimEnd<T>(this ReadOnlySpan<T> span, T trimElement)
+            where T : IEquatable<T>
+            => TrimImpl(ref span, trimElement, false, true);
+
+        /// <summary>
+        /// Conditionally removes all leading and trailing occurrences of a specified element.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="span">The source span from which the element is removed.</param>
+        /// <param name="trimElement">The specified element to look for and remove.</param>
+        /// <param name="trimStart">If set to True, trims the start.</param>
+        /// <param name="trimEnd">If set to True, trims the end.</param>
+        private static ReadOnlySpan<T> TrimImpl<T>(ref ReadOnlySpan<T> span, T trimElement, bool trimStart, bool trimEnd)
+            where T : IEquatable<T>
+        {
```

Comment:
I debated having two separate helpers, `TrimStartImpl` and `TrimEndImpl` vs the proposed `flag`ged one.
I ultimately decided on the flags because of the functional dependency that `TrimEndImpl` would have on `TrimStartImpl`:
`for (; end >= start; end--)`
So it would seem cleaner to inline them in one method.